scripts/generator/graphql_generator/csharp/SDK/Android/AndroidWebCheckout.cs.erb

```diff
@@ -0,0 +1,41 @@
+namespace <%= namespace %>.SDK.Android {
+    #if UNITY_ANDROID
+    using System;
+    using <%= namespace %>.SDK;
+
+    class AndroidWebCheckout: WebCheckout {
+
+        private ShopifyClient _client;
+        private Cart _cart;
+
+        protected override ShopifyClient Client {
+            get {
+                return _client;
+            }
+        }
+
+        protected override Cart Cart {
+            get {
+                return _cart;
+            }
+        }
+
+        private const string UnityPlayerClassName = "com.unity3d.player.UnityPlayer";
+        private const string WebCheckoutSessionClassName = "com.shopify.unity.buy.WebCheckoutSession";
+
+        public AndroidWebCheckout(Cart cart, ShopifyClient client) {
+            _cart = cart;
+            _client = client;
+        }
+
+        public void Checkout(string checkoutURL, CheckoutSuccessCallback success, CheckoutCancelCallback cancelled, CheckoutFailureCallback failure) {
+            #if !SHOPIFY_MONO_UNIT_TEST
+            using(var webCheckoutSession = new AndroidJavaObject(WebCheckoutSessionClassName)) {
```

Comment:
If we're creating a session object in Unity to start the checkout, how do we communicate back to the Android side of things after we've done the validation check from Unity? You'd need to create a new instance or call a static method, no?

It might make sense to tie the lifecycle of the Android object with that of the checkout game object that we create to communicate between Android/Unity.